intake/catalog/tests/test_parameters.py

```diff
@@ -0,0 +1,138 @@
+import os
+import pytest
+from intake.catalog.local import LocalCatalogEntry, UserParameter
+from intake.source.base import DataSource
+
+
+class NoSource(DataSource):
+
+    def __init__(self, **kwargs):
+        self.kwargs = kwargs
+
+
+driver = 'intake.catalog.tests.test_parameters.NoSource'
+
+
+def test_simplest():
+    e = LocalCatalogEntry('', '', driver, args={'arg1': 1})
+    s = e()
+    assert s.kwargs['arg1'] == 1
+
+
+def test_parameter_default():
+    up = UserParameter('name', default='oi')
+    e = LocalCatalogEntry('', '', driver, args={'arg1': "{{name}}"},
+                          parameters=[up])
+    s = e()
+    assert s.kwargs['arg1'] == 'oi'
+
+
+def test_maybe_par_env():
+    up = UserParameter('name', default='env(INTAKE_TEST_VAR)')
+    e = LocalCatalogEntry('', '', driver, args={'arg1': "{{name}}"},
+                          parameters=[up], getenv=False)
+    s = e()
+    assert s.kwargs['arg1'] == 'env(INTAKE_TEST_VAR)'
+
+    os.environ['INTAKE_TEST_VAR'] = 'oi'
+    up = UserParameter('name', default='env(INTAKE_TEST_VAR)')
+    e = LocalCatalogEntry('', '', driver, args={'arg1': "{{name}}"},
+                          parameters=[up])
+    s = e()
+    assert s.kwargs['arg1'] == 'oi'
+
+    del os.environ['INTAKE_TEST_VAR']
+
+    s = e()
+    assert s.kwargs['arg1'] == ''
+
+
+def test_up_averride():
+    up = UserParameter('name', default='env(INTAKE_TEST_VAR)')
+    e = LocalCatalogEntry('', '', driver, args={'arg1': "{{name}}"},
+                          parameters=[up], getenv=False)
+    s = e(name='other')
+    assert s.kwargs['arg1'] == 'other'
+
+
+def test_override():
+    up = UserParameter('name', default='env(INTAKE_TEST_VAR)')
+    e = LocalCatalogEntry('', '', driver, args={'arg1': "{{name}}"},
+                          parameters=[up], getenv=False)
+    s = e(arg1='other')
+    assert s.kwargs['arg1'] == 'other'
+
+
+def test_auto_env():
+    os.environ['INTAKE_TEST_VAR'] = 'oi'
+    e = LocalCatalogEntry('', '', driver,
+                          args={'arg1': "{{env(INTAKE_TEST_VAR)}}"},
+                          parameters=[], getenv=True)
+    s = e()
+    assert s.kwargs['arg1'] == 'oi'
+
+    del os.environ['INTAKE_TEST_VAR']
+
+    s = e()
+    assert s.kwargs['arg1'] == ''
+
+    e = LocalCatalogEntry('', '', driver,
+                          args={'arg1': "{{env(INTAKE_TEST_VAR)}}"},
+                          parameters=[], getenv=False)
+    s = e()
+    assert s.kwargs['arg1'] == '{{env(INTAKE_TEST_VAR)}}'
```

Comment:
I'd like to see this tested with the env var set as well

Reply:
Isn't that what happens a few lines above?